### ISSUE TITLE
RejectedExecutionException when switching project

### ISSUE THREAD

**Issue by adietish** [redhat-developer/intellij-kubernetes#103]
follow up/related to #99 
related to #112, #124

Was on the kubeadmin context where I set the openshift-pipelines project as current and expanded the Pods node.
Then switched to the crc-developer context.
Had then this in the log:

```
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@3aa2675e rejected from java.util.concurrent.ScheduledThreadPoolExecutor@639ee412[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]
	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2047)
	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:823)
	at java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:326)
	at java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:533)
	at java.util.concurrent.ScheduledThreadPoolExecutor.submit(ScheduledThreadPoolExecutor.java:632)
	at java.util.concurrent.Executors$DelegatedExecutorService.submit(Executors.java:678)
	at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager.scheduleReconnect(WatchConnectionManager.java:304)
	at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager.access$800(WatchConnectionManager.java:50)
	at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager$1.onFailure(WatchConnectionManager.java:216)
	at okhttp3.internal.ws.RealWebSocket.failWebSocket(RealWebSocket.java:571)
	at okhttp3.internal.ws.RealWebSocket$2.onResponse(RealWebSocket.java:198)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:203)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by adietish** [comment]
The stacktrace is exactly the same as in #124. The issue should be tackled in kubernetes-client at fabric8io/kubernetes-client#2624

**Comment by adietish** [comment]
this seems fixed when using kubernetes-client 5.0.0-alpha-2 (https://github.com/fabric8io/kubernetes-client/commit/fe13a511276aacf95979c3c612b9e9ad83cc5def, https://github.com/fabric8io/kubernetes-client/commit/eac6826942e002a485666c12bf1d4b3d18edbd35 and https://github.com/fabric8io/kubernetes-client/commit/1dbb5636820340c9e6417406eaba94d7c263b31e). 
We won't use an alpha version though. According to the team @kubernetes-client they'll either release a stable 5.0.0 or fork 4.x and backport the fixes. I'll thus keep this issue on hold and wait for a new release in kubernetes-client.

**Comment by adietish** [comment]
[kubernetes-client 5.0.0-beta1](https://search.maven.org/artifact/io.fabric8/kubernetes-client/5.0.0-beta-1/jar) was released on 20201221. We need a final or even better a 4.x release with the backported fix.

**Comment by adietish** [comment]
[kubernetes-client 5.0.0](https://search.maven.org/artifact/io.fabric8/kubernetes-client/5.0.0/jar) was released on 20201230. I'll test this version in intellij-tekton and verify that it would still work fine with this new client. If it does we'd be able to bump the client in intellij-kubernetes. If not we'd need a 4.x client with backported fixes.

**Comment by adietish** [comment]
Can't reproduce this after the switch to kubernetes-client 5.0.0 in #152.
Closing.
